packages/cli-kit/src/private/node/ui/hooks/use-abort-signal.ts

```diff
@@ -0,0 +1,17 @@
+import {AbortSignal} from '../../../../public/node/abort.js'
+import {useApp} from 'ink'
+import {useLayoutEffect, useState} from 'react'
+
+export default function useAbortSignal(abortSignal?: AbortSignal) {
+  const {exit: unmountInk} = useApp()
+  const [isAborted, setIsAborted] = useState(false)
+
+  useLayoutEffect(() => {
+    abortSignal?.addEventListener('abort', () => {
+      setIsAborted(true)
+      unmountInk()
```

Comment:
What would happen if I want to abort a component (e.g. a loading bar) but continue to run the CLI using Ink for with other components? Would this make Ink unusable after aborting?